src/coreclr/pal/src/include/pal/context.h

```diff
@@ -33,9 +33,7 @@ extern "C"
 /* A type to wrap the native context type, which is ucontext_t on some
  * platforms and another type elsewhere. */
 #if HAVE_UCONTEXT_T
-#if HAVE_UCONTEXT_H
-#include <ucontext.h>
-#endif // HAVE_UCONTEXT_H
+#include <sys/ucontext.h>
```

Comment:
This is basically reverting https://github.com/dotnet/runtime/commit/796cbeeec0a97428e97b92250cb9ff65aadf41de. ucontext.h is POSIX while sys/ucontext.h is platform detail. Please put it under `TARGET_OSX` if it is only needed on mac.

Reply:
Hmm, I suppose you are technically correct. We actually have some unguarded usages of `sys/ucontext.h` and few ones that are guarded with `HAVE_SYS_UCONTEXT_H`. That convinced me that all supported platforms actually have `sys/ucontext.h`.

This originally started off as an attempt to avoid defining `_XOPEN_SOURCE`. Weirdly enough the commit you referenced should have made it defined everywhere, yet I had build failures related to it, and it's still redefined in NativeAOT and Mono CMakeFiles.

Context: https://github.com/dotnet/runtime/pull/109928#discussion_r1850957729

Reply:
Ah, the current version of the code explains the bit:
```
elseif(CLR_CMAKE_HOST_OSX AND NOT CLR_CMAKE_HOST_MACCATALYST AND NOT CLR_CMAKE_HOST_IOS AND NOT CLR_CMAKE_HOST_TVOS)
  add_definitions(-D_XOPEN_SOURCE)
```

I wonder why the mobile platforms were excluded.

Reply:
https://github.com/dotnet/runtime/pull/110092. Let's check that first. If it gets accepted then I can revert back to `<ucontext.h>`.

Reply:
Rebased, reverted and simplified.